src/machine/CoffeeMachine.java

```diff
@@ -0,0 +1,80 @@
+package machine;
+
+class CoffeeMachine {
+
+    private final CoffeeMachineSupplies coffeeMachineSupplies;
+
+    // For our purposes it is sufficient that the price has int type
+    private int dollars;
+
+    private CoffeeMachineState currentState;
+
+    CoffeeMachineState getCurrentState() {
+        return currentState;
+    }
+
+    CoffeeMachine(CoffeeMachineSupplies coffeeMachineSupplies, int dollars) {
+        this.coffeeMachineSupplies = coffeeMachineSupplies;
+        this.dollars = dollars;
+        this.currentState = CoffeeMachineState.CHOOSING_AN_ACTION;
+    }
+
+    private void displayContent() {
+        System.out.println("\nThe coffee machine has:");
+        coffeeMachineSupplies.displaySupplies();
+        System.out.printf("$%d of money\n", dollars);
+    }
+
+    private void prepareCoffee(int chosenOption) {
+        final var coffee = Coffee.fromOption(chosenOption);
+        try {
+            coffeeMachineSupplies.reduceSupplies(coffee);
+            dollars += coffee.getPriceInDollars();
+            System.out.println();
+        } catch (IllegalStateException exception) {}
```

Comment:
Should we log here that the coffee cannot be prepared?

Reply:
We already have output for user for every case in `canMake(Coffee coffee)` inside `coffeeMachineSupplies.reduceSupplies(coffee)`